freeipa/src/main/java/com/sequenceiq/freeipa/kerberosmgmt/v1/KerberosMgmtV1Service.java

```diff
@@ -181,4 +192,30 @@ private boolean isDuplicateEntryException(FreeIpaClientException e) {
                 .filter(c -> c == DUPLICATE_ENTRY_ERROR_CODE)
                 .isPresent();
     }
+
+    SecretResponse getSecretResponseForPrincipal(String accountId, String principal) {
+        try {
+            String path = String.format("%s/%s/%s/%s-%s", accountId,
+                    "ServiceKeytab", "serviceprincipal",
+                    UUID.randomUUID().toString(), Long.toHexString(System.currentTimeMillis()));
```

Comment:
Why is a timestamp needed? Doesn't the UUID already handle the time for uniqueness (ie if the UUID is a duplicate then the timestamp will also be the same) or is there timestamp useful for something else?

(Same for `getSecretResponseForKeytab()`)

Reply:
This is the convention followed in other places where the vault is used. I followed it.